File: book_rental_frontend/src/app/service/language.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import base_url from './base_url';

@Injectable({
  providedIn: 'root'
})
export class LanguageService {

  constructor(private _http:HttpClient) { }

  //Load all languages
  public getLanguage() {
    return this._http.get(`${base_url}/language`);
  }

  //Add Language
  public addLanguage(language:any){
    return this._http.post(`${base_url}/language`, language);
  }

  //Delete Language
  public deleteLanguage(lang_id:any){
    return this._http.delete(`${base_url}/language/${lang_id}`);
  }

  //Update language
}

File: book_rental_server/src/main/java/com/bookrental/service/LanguageService.java
package com.bookrental.service;

import java.util.List;

import com.bookrental.model.Language;

public interface LanguageService {

	//Add Language
	public Language addLanguage(Language language);
	//Get All Language
	public List<Language> getAllLanguage();
	//Get Specific Language Using id
//	public Language getLanguage(Long id);
	//Update specific language using id
//	public Language updateLanguage(Long id, Language language);
	//Delete specific Language using id
	public void deleteLanguage(Long id);
}
